build/setup.pm

```diff
@@ -132,10 +132,11 @@ our %TC_POSIX = (
     ldsys => undef,
     ldimp => undef,
 
-    ccshared   => '-fPIC',
-    ldshared   => '-shared @ccshared@',
-    moarshared => '',
-    ldrpath    => '-Wl,-rpath,"/@libdir@"',
+    ccshared            => '-fPIC',
+    ldshared            => '-shared @ccshared@',
+    moarshared          => '',
+    ldrpath             => '-Wl,-rpath,"/@libdir@"',
+    ldrpath_relocatable => '-Wl,-z,origin,-rpath,\'$$ORIGIN/../lib\'',
```

Comment:
I don't think that the `-z` flag to `ld` is terribly portable. It looks like the macOS linker is not a fan of that flag.